.travis.yml

```diff
@@ -54,15 +60,42 @@ jobs:
         - *snapcraft
         - review-tools
       env: BUILD_TYPE=RelWithDebInfo
+      before_script: &before_snapcraft
+      # patch build system for the build type being processed
+      - "[ ! -f tests/travis.patch ] || patch -p1 --no-backup-if-mismatch < tests/travis.patch"
+      - "[ ! -f tests/travis-${BUILD_TYPE}.patch ] || patch -p1 --no-backup-if-mismatch < tests/travis-${BUILD_TYPE}.patch"
+
+      # set up ccache
+      - sg lxd -c '/snap/bin/lxc profile set default environment.PATH "/usr/lib/ccache:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin"'
+      - sg lxd -c '/snap/bin/lxc profile device add default ccache disk source=${HOME}/.ccache/ path=/root/.ccache'
+      - ccache --zero-stats
+
+      # inject build identifier
+      - sed -i "/configflags:/a \    - -DMULTIPASS_BUILD_LABEL=${MULTIPASS_BUILD_LABEL}" snap/snapcraft.yaml
+
       script:
       - sg lxd -c '/snap/bin/snapcraft --use-lxd'
       - /snap/bin/review-tools.snap-review --allow-classic *.snap
+      after_success:
+      - '[[ "${TRAVIS_EVENT_TYPE}" != "pull_request" || "${TRAVIS_SECURE_ENV_VARS}" == "false" ]] && exit'
+      - BUILDS=()
+      - >-
+        [ -n "${MULTIPASS_SNAP_CHANNEL}" ]
+        && snapcraft login --with ${HOME}/snap-login
+        && snapcraft push multipass_*.snap --release ${MULTIPASS_SNAP_CHANNEL}
+        && BUILDS+=("\`snap refresh multipass --channel ${MULTIPASS_SNAP_CHANNEL}\`")
+      - CURL_OUTPUT=$( timeout 5m curl --fail --upload-file *.snap https://transfer.sh/ )
+        && BUILDS+=("[$( basename ${CURL_OUTPUT} )](${CURL_OUTPUT})");
+        echo ${CURL_OUTPUT}
+      - ./tools/report_build.py Snap "${BUILDS[@]}"
```

Comment:
I now wonder whether we should do this on the Debug build? The snap checks and now the uploads mean that this job runs almost twice as long as the Debug one.

And maybe it'd be better for debugging to have Debug builds out of PRs?

Reply:
I don't know, I don't have a strong opinion on that. Release would be closer to the actual thing, but debug would indeed be more practical. Would that affect actual releasing? And aren't the snap checks useful in the build that uploads the snap? What is the gain in having debug and release builds switching places in what concerns duration?

Reply:
> Would that affect actual releasing?

No, for releases (RC we'd have to decide on) we'd build RelWithDebInfo still.

> And aren't the snap checks useful in the build that uploads the snap?

At worst they would fail to upload. Now I think of it maybe we can drop the review step as the store does it for us… The snap upload would have to move up to `script:` though, so failure to release would be a CI failure, and alternate with `review-tools` if not publishing.

> What is the gain in having debug and release builds switching places in what concerns duration?

One would do the snap review check, another would do the upload.

Any case, something to ponder for the future.

Reply:
Alright

Reply:
At least the "faster" concern is moot now: #1141